src/components/implementation/no_interface/scheddev/sl_lock.c

```diff
@@ -0,0 +1,67 @@
+#include "sl.h"
+#include "sl_lock.h"
+
+void
+sl_lock_init(struct sl_lock *lock)
+{
+    assert(lock);
+    *lock = SL_LOCK_STATIC_INIT();
+}
+
+thdid_t
+sl_lock_holder(struct sl_lock *lock)
+{
+    assert(lock);
```

Comment:
Shouldn't `assert` be compiled out in production code? I'm happy to remove these if that is not the case.